packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -202,9 +238,163 @@ export default async function initSanity(
+
+      const embeddedStudioPath = path.join(
+        srcPath,
+        `${studioPath}/`,
+        useAppDir ? '[[...index]]/page.tsx' : '[[...index]].tsx'
+      )
+
+      // this selects the correct template string based on whether the user is using the app or pages directory and
+      // replaces the ":configPath:" placeholder in the template with the correct path to the sanity.config.ts file.
+      // we account for the user-defined embeddedStudioPath (default /studio) is accounted for by creating enough "../"
+      // relative paths to reach the root level of the project
+      await writeOrOverwrite(
+        embeddedStudioPath,
+        (useAppDir ? sanityStudioAppTemplate : sanityStudioPagesTemplate).replace(
+          ':configPath:',
+          new Array(embeddedStudioPath.slice(workDir.length).split('/').length - 1)
+            .join('../')
+            .concat('sanity.config')
+        )
+      )
+
+      const sanityConfigPath = path.join(workDir, 'sanity.config.'.concat(fileEnding))
+      await writeOrOverwrite(
+        sanityConfigPath,
+        sanityConfigTemplate
+          .replace(':route:', embeddedStudioPath.slice(workDir.length).replace('src/', ''))
+          .replace(':basePath:', studioPath)
+      )
+    }
+
+    const sanityCliPath = path.join(workDir, 'sanity.cli.'.concat(fileEnding))
+    await writeOrOverwrite(sanityCliPath, sanityCliTemplate)
+
+    // write sanity folder files
+    const recursivelyWriteFiles = async (
+      files: Record<string, string | Record<string, string>>,
+      folderPath?: string
+    ) => {
+      for (const [filePath, content] of Object.entries(files)) {
+        if (filePath.includes('.') && typeof content === 'string') {
+          await writeOrOverwrite(
+            path.join(workDir, 'sanity', folderPath || '', filePath.concat(fileEnding)),
+            content
+          )
+        } else {
+          await mkdirSync(path.join(workDir, 'sanity', filePath), {recursive: true})
+          if (typeof content === 'object') {
+            await recursivelyWriteFiles(content, filePath)
+          }
+        }
+      }
+    }
+
+    // ask what kind of schema setup the user wants
+    const templateToUse = await promptForNextTemplate(prompt)
+
+    const spinner = await recursivelyWriteFiles(sanityFolder(useTypeScript, templateToUse)).then(
+      () => {
+        return context.output.spinner('Installing Sanity dependencies').start()
+      }
+    )
+
+    // set tsconfig.json target to ES2017
+    const tsConfigPath = path.join(workDir, 'tsconfig.json')
+    if (useTypeScript && existsSync(tsConfigPath)) {
+      const tsConfig = JSON.parse(readFileSync(tsConfigPath, 'utf8'))
+      tsConfig.compilerOptions.target = 'ES2017'
+      await writeFileSync(tsConfigPath, JSON.stringify(tsConfig, null, 2))
+    }
+
+    // install sanity deps
+    await exec('npm install next-sanity@4 @sanity/vision@3 sanity@3 @sanity/image-url@1', {
```

Comment:
Preferably use either the method I mentioned further down, or use the execa dependency for a slightly more elegant execution?